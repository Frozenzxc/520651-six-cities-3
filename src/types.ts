export enum OfferType {
    ALL = `cities`,
    FAVORITES = `favorites`,
    NEARBY = `near`,
}

export enum SortType {
    POPULAR = `Popular`,
    PRICE_TO_HIGH = `Price: low to high`,
    PRICE_TO_LOW = `Price: high to low`,
    RATING = `Top rated first`,
}

export interface Offer {
    bedrooms: number,
    city: {
        location: {
            latitude: number,
            longitude: number,
            zoom: number,
            },
        name: string,
    },
    description: string,
    goods: string[],
    host: {
        avatar: string,
        id: number,
        isPro: boolean,
        name: string,
     },
    id: number,
    images: string[],
    isFavorite: boolean,
    isPremium: boolean,
    location: {
        latitude: number,
        longitude: number,
        zoom: number,
     },
    maxAdults: number,
    previewImage: string,
    price: number,
    rating: number,
    title: string,
    type: string,
}

export interface Review {
    comment: string,
    date: string,
    id: number,
    rating: number,
    user: {
        avatar: string,
        id: number,
        isPro: boolean,
        name: string,
    },
}
