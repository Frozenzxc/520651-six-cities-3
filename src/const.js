const OfferType = {
  ALL: `cities`,
  FAVORITES: `favorites`,
  NEARBY: `near`,
};

const MAX_NEARBY_OFFERS = 3;

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const ReviewPostingStatus = {
  POSTED: `POSTED`,
  ERROR: `ERROR`,
};

const AppRoute = {
  FAVORITES: `/favorites`,
  LOGIN: `/login`,
  PROPERTY: `/offer/:id`,
  ROOT: `/`,
};

export {MAX_NEARBY_OFFERS, AuthorizationStatus, AppRoute, ReviewPostingStatus, OfferType};
