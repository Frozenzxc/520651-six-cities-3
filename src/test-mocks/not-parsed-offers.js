/* eslint-disable camelcase */
export default [
  {
    bedrooms: 3,
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.938361,
        longitude: 6.959974,
        zoom: 13
      }
    },
    description: `Design interior in most sympathetic area! Complitely renovated, well-equipped, cosy studio in idyllic, over 100 years old wooden house. Calm street, fast connection to center and airport.`,
    goods: [
      `Coffee machine`,
      `Laptop friendly workspace`,
      `Fridge`,
      `Breakfast`,
      `Air conditioning`,
      `Towels`,
      `Washer`,
      `Dishwasher`,
      `Baby seat`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar_url: `img/avatar-angelina.jpg`,
      is_pro: true
    },
    id: 1,
    images: [
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`
    ],
    is_favorite: false,
    is_premium: false,
    location: {
      latitude: 50.932361,
      longitude: 6.937974,
      zoom: 16
    },
    max_adults: 7,
    preview_image: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg`,
    price: 160,
    rating: 4.8,
    title: `The house among olive `,
    type: `apartment`
  },
  {
    bedrooms: 1,
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.938361,
        longitude: 6.959974,
        zoom: 13
      }
    },
    description: `Design interior in most sympathetic area! Complitely renovated, well-equipped, cosy studio in idyllic, over 100 years old wooden house. Calm street, fast connection to center and airport.`,
    goods: [
      `Breakfast`,
      `Washer`,
      `Air conditioning`,
      `Laptop friendly workspace`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar_url: `img/avatar-angelina.jpg`,
      is_pro: true
    },
    id: 2,
    images: [
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg`
    ],
    is_favorite: false,
    is_premium: false,
    location: {
      latitude: 50.913361,
      longitude: 6.9509739999999995,
      zoom: 16
    },
    max_adults: 1,
    preview_image: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg`,
    price: 192,
    rating: 4.2,
    title: `Waterfront with extraordinary view`,
    type: `room`
  },
  {
    bedrooms: 3,
    city: {
      name: `Paris`,
      location: {
        latitude: 48.85661,
        longitude: 2.351499,
        zoom: 13
      }
    },
    description: `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
    goods: [
      `Washer`,
      `Baby seat`,
      `Laptop friendly workspace`,
      `Breakfast`,
      `Air conditioning`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar_url: `img/avatar-angelina.jpg`,
      is_pro: true
    },
    id: 3,
    images: [
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg`
    ],
    is_favorite: false,
    is_premium: false,
    location: {
      latitude: 48.87961000000001,
      longitude: 2.353499,
      zoom: 16
    },
    max_adults: 4,
    preview_image: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg`,
    price: 350,
    rating: 4.4,
    title: `Nice, cozy, warm big bed apartment`,
    type: `hotel`
  },
  {
    bedrooms: 2,
    city: {
      name: `Cologne`,
      location: {
        latitude: 50.938361,
        longitude: 6.959974,
        zoom: 13
      }
    },
    description: `Design interior in most sympathetic area! Complitely renovated, well-equipped, cosy studio in idyllic, over 100 years old wooden house. Calm street, fast connection to center and airport.`,
    goods: [
      `Breakfast`,
      `Laptop friendly workspace`,
      `Washer`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar_url: `img/avatar-angelina.jpg`,
      is_pro: true
    },
    id: 4,
    images: [
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg`
    ],
    is_favorite: false,
    is_premium: false,
    location: {
      latitude: 50.949361,
      longitude: 6.976974,
      zoom: 16
    },
    max_adults: 3,
    preview_image: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg`,
    price: 196,
    rating: 3.3,
    title: `Loft Studio in the Central Area`,
    type: `house`
  },
  {
    bedrooms: 2,
    city: {
      name: `Hamburg`,
      location: {
        latitude: 53.550341,
        longitude: 10.000654,
        zoom: 13
      }
    },
    description: `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
    goods: [
      `Breakfast`,
      `Washer`,
      `Towels`,
      `Air conditioning`,
      `Laptop friendly workspace`,
      `Baby seat`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar_url: `img/avatar-angelina.jpg`,
      is_pro: true
    },
    id: 5,
    images: [
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/14.jpg`
    ],
    is_favorite: false,
    is_premium: true,
    location: {
      latitude: 53.563341,
      longitude: 9.975654,
      zoom: 16
    },
    max_adults: 7,
    preview_image: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/7.jpg`,
    price: 116,
    rating: 4.2,
    title: `Penthouse, 4-5 rooms + 5 balconies`,
    type: `apartment`
  },
  {
    bedrooms: 5,
    city: {
      name: `Brussels`,
      location: {
        latitude: 50.846557,
        longitude: 4.351697,
        zoom: 13
      }
    },
    description: `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
    goods: [
      `Fridge`,
      `Air conditioning`,
      `Breakfast`,
      `Baby seat`,
      `Coffee machine`,
      `Towels`,
      `Laptop friendly workspace`,
      `Washer`,
      `Dishwasher`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar_url: `img/avatar-angelina.jpg`,
      is_pro: true
    },
    id: 6,
    images: [
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/17.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg`
    ],
    is_favorite: false,
    is_premium: false,
    location: {
      latitude: 50.869557,
      longitude: 4.332697,
      zoom: 16
    },
    max_adults: 9,
    preview_image: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg`,
    price: 892,
    rating: 3,
    title: `Loft Studio in the Central Area`,
    type: `house`
  },
  {
    bedrooms: 2,
    city: {
      name: `Hamburg`,
      location: {
        latitude: 53.550341,
        longitude: 10.000654,
        zoom: 13
      }
    },
    description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
    goods: [
      `Laptop friendly workspace`,
      `Towels`,
      `Air conditioning`,
      `Washer`,
      `Breakfast`,
      `Baby seat`,
      `Fridge`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar_url: `img/avatar-angelina.jpg`,
      is_pro: true
    },
    id: 7,
    images: [
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/10.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/9.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/6.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/11.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg`
    ],
    is_favorite: false,
    is_premium: false,
    location: {
      latitude: 53.528341000000005,
      longitude: 10.018654000000002,
      zoom: 16
    },
    max_adults: 9,
    preview_image: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/19.jpg`,
    price: 136,
    rating: 5,
    title: `Waterfront with extraordinary view`,
    type: `apartment`
  },
];
