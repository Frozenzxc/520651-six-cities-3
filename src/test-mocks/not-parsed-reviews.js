/* eslint-disable camelcase */
export default [
  {
    id: 1,
    user: {
      id: 17,
      is_pro: false,
      name: `Emely`,
      avatar_url: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/8.jpg`
    },
    rating: 4,
    comment: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!`,
    date: `2020-02-23T08:36:26.433Z`
  },
  {
    id: 2,
    user: {
      id: 14,
      is_pro: true,
      name: `Corey`,
      avatar_url: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/5.jpg`
    },
    rating: 3,
    comment: `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
    date: `2020-02-23T08:36:26.433Z`
  }
];
