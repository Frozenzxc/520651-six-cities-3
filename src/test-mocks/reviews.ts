import {Review} from "../types";

export const reviews: Review[] = [
  {
    comment: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!`,
    date: `2020-02-23T08:36:26.433Z`,
    id: 1,
    rating: 4,
    user: {
      avatar: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/8.jpg`,
      id: 17,
      isPro: false,
      name: `Emely`
    }
  },
  {
    comment: `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
    date: `2020-02-23T08:36:26.433Z`,
    id: 2,
    rating: 3,
    user: {
      avatar: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/5.jpg`,
      id: 14,
      isPro: true,
      name: `Corey`
    }
  }
];
