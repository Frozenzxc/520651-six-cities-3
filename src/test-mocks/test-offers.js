export default [
  {
    bedroomsCount: 4,
    coordinates: [52.3909553943508, 4.85309666406198],
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
    host: {
      avatar: `img/avatar-max.jpg`,
      name: `John`,
      super: false,
    },
    id: 1,
    maxGuests: 6,
    options: [`Wifi`, `Washing machine`, `Towels`, `Heating`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
    premium: false,
    price: 200,
    rating: 3,
    reviews: [
      {
        date: `2020-02-22T00:00:00.000Z`,
        reviewID: 140,
        name: `Max`,
        rating: 1,
        src: `img/avatar-max.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
      {
        date: `2020-02-24T00:00:00.000Z`,
        reviewID: 151,
        name: `John`,
        rating: 4,
        src: `img/avatar-max.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
      {
        date: `2020-02-26T00:00:00.000Z`,
        reviewID: 160,
        name: `Victoria`,
        rating: 3,
        src: `img/avatar-max.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
      {
        date: `2020-03-02T00:00:00.000Z`,
        reviewID: 145,
        name: `Bob`,
        rating: 5,
        src: `img/avatar-max.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
    ],
    src: [`img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
    title: `Beautiful & luxurious apartment at great location`,
    type: `house`,
  },
  {
    bedroomsCount: 2,
    coordinates: [52.369553943508, 4.85309666406198],
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Max`,
      super: false,
    },
    id: 2,
    maxGuests: 4,
    options: [`Wifi`, `Towels`, `Heating`, `Cabel TV`, `Fridge`],
    premium: true,
    price: 300,
    rating: 1,
    reviews: [
      {
        date: `2020-02-26T00:00:00.000Z`,
        reviewID: 156,
        name: `Alex`,
        rating: 4,
        src: `img/avatar-angelina.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
      {
        date: `2020-02-24T00:00:00.000Z`,
        reviewID: 198,
        name: `Dana`,
        rating: 1,
        src: `img/avatar-angelina.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
    ],
    src: [`img/apartment-02.jpg`, `img/apartment-03.jpg`],
    title: `Wood and stone place`,
    type: `hotel`,
  },
  {
    bedroomsCount: 1,
    coordinates: [52.3909553943508, 4.929309666406198],
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
    host: {
      avatar: `img/avatar-max.jpg`,
      name: `Bob`,
      super: true,
    },
    id: 3,
    maxGuests: 2,
    options: [`Wifi`, `Towels`, `Cabel TV`],
    premium: false,
    price: 450,
    rating: 3,
    reviews: [
      {
        date: `2020-02-26T00:00:00.000Z`,
        reviewID: 176,
        name: `Max`,
        rating: 2,
        src: `img/avatar-angelina.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
    ],
    src: [`img/apartment-03.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`],
    title: `Canal View Prinsengracht`,
    type: `room`,
  },
  {
    bedroomsCount: 2,
    coordinates: [52.3809553943508, 4.939309666406198],
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      super: false,
    },
    id: 4,
    maxGuests: 5,
    options: [`Wifi`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Cabel TV`],
    premium: false,
    price: 150,
    rating: 5,
    reviews: [
      {
        date: `2020-02-26T00:00:00.000Z`,
        reviewID: 169,
        name: `John`,
        rating: 4,
        src: `img/avatar-max.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
      {
        date: `2020-02-24T00:00:00.000Z`,
        reviewID: 183,
        name: `Victoria`,
        rating: 1,
        src: `img/avatar-max.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
      {
        date: `2020-03-02T00:00:00.000Z`,
        reviewID: 147,
        name: `Victoria`,
        rating: 3,
        src: `img/avatar-max.jpg`,
        text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.`,
      },
    ],
    src: [`img/apartment-03.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`],
    title: `Nice, cozy, warm big bed apartment`,
    type: `apartment`,
  },
];
