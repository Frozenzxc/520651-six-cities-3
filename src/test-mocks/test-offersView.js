const OffersView = {
  ALL: `cities__`,
  NEARBY: `near-`,
};

export {OffersView};
