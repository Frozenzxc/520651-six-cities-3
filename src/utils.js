export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const parseOffer = (data) => {
  let offer = {};
  offer.bedrooms = data[`bedrooms`];
  offer.city = data[`city`];
  offer.description = data[`description`];
  offer.goods = data[`goods`];
  offer.host = {};
  offer.host.avatar = data[`host`][`avatar_url`];
  offer.host.id = data[`host`][`id`];
  offer.host.isPro = data[`host`][`is_pro`];
  offer.host.name = data[`host`][`name`];
  offer.id = data[`id`];
  offer.images = data[`images`];
  offer.isFavorite = data[`is_favorite`];
  offer.isPremium = data[`is_premium`];
  offer.location = data[`location`];
  offer.maxAdults = data[`max_adults`];
  offer.previewImage = data[`preview_image`];
  offer.price = data[`price`];
  offer.rating = data[`rating`];
  offer.title = data[`title`];
  offer.type = data[`type`];

  return offer;
};

export const parseReview = (data) => {
  let review = {};
  review.comment = data[`comment`];
  review.date = data[`date`];
  review.id = data[`id`];
  review.rating = data[`rating`];
  review.user = {};
  review.user.avatar = data[`user`][`avatar_url`];
  review.user.id = data[`user`][`id`];
  review.user.isPro = data[`user`][`is_pro`];
  review.user.name = data[`user`][`name`];

  return review;
};

export const getCitiesList = (offers) => {
  return [...new Set(offers.map((offer) => offer.city.name))];
};

export const noop = () => {};
