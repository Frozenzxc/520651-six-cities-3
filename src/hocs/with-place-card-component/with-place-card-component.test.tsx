import * as React from "react";
import * as renderer from "react-test-renderer";
import {AuthorizationStatus} from "../../const";
import withPlaceCardComponent from "./with-place-card-component";
import {offers} from "../../test-mocks/test-offers";
import {OfferType} from "../../types";
import {noop} from "../../utils";

const offer = offers[0];

const MockComponent = () => {

  return (
    <div>
    </div>
  );
};

const MockComponentWrapped = withPlaceCardComponent(MockComponent);

it(`withPlaceCardComponent is rendered correctly`, () => {

  const tree = renderer.create((
    <MockComponentWrapped
      addToFavorite={noop}
      authorizationStatus={AuthorizationStatus.NO_AUTH}
      isFavorite={true}
      offer={offer}
      onClick={noop}
      onMouseEnter={noop}
      onCardHover={noop}
      onFavoriteClick={noop}
      onTitleClick={noop}
      offersView={OfferType.ALL}
    >
    </MockComponentWrapped>
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
