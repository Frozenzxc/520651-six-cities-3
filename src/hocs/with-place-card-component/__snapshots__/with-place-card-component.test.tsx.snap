// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withPlaceCardComponent is rendered correctly 1`] = `<div />`;
